guides/source/configuring.md

```diff
@@ -1267,6 +1268,39 @@ The default value depends on the `config.load_defaults` target version:
 | (original)            | `false`              |
 | 7.0                   | `true`               |
 
+#### `config.active_record.rollback_transaction_on_non_local_return`
+
+Defines whether `return`, `break` and `throw` inside a `transaction` block cause the transaction to be
+committed or rolled back. e.g.:
+
+```ruby
+Model.transaction do
+  model.save
+  return
+  other_model.save # not executed
+end
+```
+
+If set to `true`, the above transaction will be rolled back.
+
+If set to `false`, it will be committed.
+
+| Starting with version | The default value is |
+| --------------------- | -------------------- |
+| (original)            | `true`               |
+| 7.1                   | `false`              |
+
+Historically only raised errors would trigger a rollback, but in Ruby `2.3`, the `timeout` library
+started using `throw` to interupt execution which had the adverse effect of committing open transactions.
+
+To solve this, in Active Record 6.1 the behavior was changed to instead rollback the transaction as it was safer
+than to potentially commit an incomplete transaction.
```

Comment:
IIRC technically it was deprecated in 6.1, changed in 7.0.

Overall this description feels a bit more changeloggy than config-explainy.

Assuming we're immediately deprecating the rollback behaviour, I think we only need to express 1) this is used to give new 7.1 applications the commit behaviour; 2) existing applications _may_ set this option, noting they'll miss any deprecation and immediately invert behaviour of any non-local returns; 3) the rollback behaviour is deprecated and will be removed in 7.2.

(For usable upgrades, and because we _want_ people to have the proper opportunity to find deprecated usage of rollback, we want to deprecate _hitting_ rollback, not _having the option set to rollback mode_ -- so although we'll remove the rollback _behaviour_ in 7.2, it'll only be at that point that we deprecate this config option, for the handful of people who've set it manually.)

Reply:
> Overall this description feels a bit more changeloggy than config-explainy.

yeah for sure. I guess I felt obligated to explain why it flip-floped this much.

I'll add the deprecations.